client/app/components/users/UserEdit.jsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import Button from 'antd/lib/button';
+import Icon from 'antd/lib/icon';
+import Input from 'antd/lib/input';
+import Tooltip from 'antd/lib/tooltip';
+import Modal from 'antd/lib/modal';
+import { react2angular } from 'react2angular';
+import { User } from '@/services/user';
+import { currentUser } from '@/services/auth';
+import { UserProfile } from '../proptypes';
+import { DynamicForm } from '../dynamic-form/DynamicForm';
+
+export class UserEdit extends React.Component {
+  static propTypes = {
+    user: UserProfile.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = { user: this.props.user, changePassword: false };
+  }
+
+  handleSave = (values, onSuccess, onError) => {
+    const data = {
+      id: this.props.user.id,
+      ...values,
+    };
+
+    User.save(data, (user) => {
+      onSuccess('Saved.');
+      this.setState({
+        user: {
+          id: user.id,
+          name: user.name,
+          email: user.email,
+          profileImageUrl: user.profile_image_url,
+          apiKey: user.api_key,
+          isDisabled: user.is_disabled,
+        },
+      });
+    }, (error) => {
+      onError(error.data.message || 'Failed saving.');
+    });
+  };
+
+  openChangePasswordModal = () => {
+    this.setState({ changePassword: true });
+  };
+
+  changePasswordModal() {
+    const fields = [
+      { name: 'old_password', title: 'Current Password' },
+      { name: 'password', title: 'New Password' },
+      { name: 'password_repeat', title: 'Repeat New Password' },
+    ].map(field => ({ ...field, type: 'password', required: true }));
+
+    return (
+      <Modal
+        visible={this.state.changePassword}
+        title="Change Password"
+        onCancel={() => { this.setState({ changePassword: false }); }}
+        footer={null}
+        destroyOnClose
+      >
+        <DynamicForm fields={fields} saveText="Update Password" onSubmit={this.handleSave} />
```

Comment:
Interesting. I never thought we will use `DynamicForm` to render forms in the app (that are not based on backend properties). But I guess it makes sense... I would draw the line in cases where rendering with DynamicForm requires too many tweaks and it's better to use the components directly.

Reply:
My thoughts are that components are a concept to avoid reuse of code. Forms can be pretty annoying to handle, so I tried to make `DynamicForm` in a way it would be an easy-to-use Object-To-Form component.

As for backend properties, it currently needs some mapping (improving how this is done is on my plans when migrating Data Sources & Destinations Page).

The biggest problem I see is that it's a lot easier to break a page you were not expecting to, and that's where creating tests take part :slightly_smiling_face: